api/v1alpha1/httproutefilter_types.go

```diff
@@ -53,6 +59,18 @@ const (
 	RegexHTTPPathModifier HTTPPathModifierType = "ReplaceRegexMatch"
 )
 
+// HTTPPathModifierType defines the type of Hostname rewrite.
+type HTTPHostnameModifierType string
+
+const (
+	// HeaderHTTPHostnameModifier indicates that the Host header value would be replaced with the value of the header specified in setFromHeader.
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-host-rewrite-header
+	HeaderHTTPHostnameModifier HTTPHostnameModifierType = "SetFromHeader"
+	// BackendHTTPHostnameModifier indicates that the Host header value would be replaced by the DNS name of the backend if it exists.
+	// https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-routeaction-auto-host-rewrite
+	BackendHTTPHostnameModifier HTTPHostnameModifierType = "SetFromBackend"
```

Comment:
do we need Backend in the first iteration ? can we only start with header ?

Reply:
this should be equivalent to the nginx ingress `nginx.ingress.kubernetes.io/upstream-vhost: $proxy_host` that allows the value of the host header to by dynamically rewritten to that of the target host. Prefer to support this if possible.